internal/backends/gl/glwindow.rs

```diff
@@ -280,17 +280,40 @@ impl WinitWindow for GLWindow {
     }
 
     fn set_icon(&self, icon: corelib::graphics::Image) {
-        if let Some(rgba) = crate::IMAGE_CACHE
-            .with(|c| c.borrow_mut().load_image_resource((&icon).into()))
-            .and_then(|i| i.to_rgba())
-        {
-            let (width, height) = rgba.dimensions();
-            if let Some(window) = self.borrow_mapped_window() {
-                window.opengl_context.window().set_window_icon(
-                    winit::window::Icon::from_rgba(rgba.into_raw(), width, height).ok(),
-                );
-            }
+        let image_inner: &ImageInner = (&icon).into();
+        let pixel_buffer = match image_inner {
+            ImageInner::EmbeddedImage { buffer, .. } => buffer.clone(),
+            _ => return,
+        };
+
+        let dynamic_image = match pixel_buffer {
+            SharedImageBuffer::RGB8(pixels) => image::DynamicImage::ImageRgb8(
+                image::RgbImage::from_raw(
+                    pixels.width(),
+                    pixels.height(),
+                    pixels.as_bytes().to_vec(),
+                )
+                .unwrap(),
+            ),
+            SharedImageBuffer::RGBA8(pixels) => image::DynamicImage::ImageRgba8(
+                image::RgbaImage::from_raw(
+                    pixels.width(),
+                    pixels.height(),
+                    pixels.as_bytes().to_vec(),
+                )
+                .unwrap(),
+            ),
+            SharedImageBuffer::RGBA8Premultiplied(_) => return, // TODO: divide out alpha
```

Comment:
Looks suspicious.  Don't we need this?

Reply:
The API allows for this, doesn’t it? I could create an rgba premultiplied image in Rust and feed it to the Window’s icon property and we would end up here.

IMO this is a valid todo.